robot_arm_driver:
  ros__parameters:
    serial_port: /dev/ttyUSB0
    baud_rate: 1000000
    revolute_joint_names: ['pan_joint', 'shoulder_joint', 'elbow_joint', 'wrist0_joint', 'wrist1_joint', 'wrist2_joint']
    initial_pos: [0.0, -0.7853, 2.35619, 0.0, 1.5708, 0.0]

    revolute_joints:
      pan_joint:
        index: 0
        id: 10
        invert: False
        min_pos: -3.1416
        max_pos: 3.1416
        angle_resolution: 0.0879 #degrees
        step_per_sec: 50
        speed: 2.0
        acceleration: 0.46
        torque: 40
        max_torque: 40
        offset: 0.0
        offset_gain: 0.0
        
      shoulder_joint:
        index: 1
        id: 11
        invert: False
        min_pos: -1.8515
        max_pos: 2.1015
        angle_resolution: 0.0879 #degrees
        step_per_sec: 1.33
        speed: 2.0
        acceleration: 0.46
        torque: 85
        max_torque: 85
        offset: 0.0122718463
        offset_gain: 10.0

      elbow_joint:
        index: 2
        id: 12
        invert: False
        min_pos: -0.4
        max_pos: 3.0
        angle_resolution: 0.0879 #degrees
        step_per_sec: 1.33
        speed: 2.0
        acceleration: 0.46
        torque: 85
        max_torque: 85
        offset: 0.00409061543
        offset_gain: 10.0

      wrist0_joint:
        index: 3
        id: 13
        invert: False
        min_pos: -3.1416
        max_pos: 3.1416
        angle_resolution: 0.0879 #degrees
        step_per_sec: 50
        speed: 2.0
        acceleration: 0.46
        torque: 40
        max_torque: 40
        offset: 0.0
        offset_gain: 0.0

      wrist1_joint:
        index: 4
        id: 14
        invert: False
        min_pos: -1.8
        max_pos: 2.0
        angle_resolution: 0.0879 #degrees
        step_per_sec: 50
        speed: 2.0
        acceleration: 0.46
        torque: 40
        max_torque: 40
        offset: 0.0
        offset_gain: 0.0

      wrist2_joint:
        index: 5
        id: 15
        invert: False
        min_pos: -3.1415
        max_pos: 3.1416
        angle_resolution: 0.0879 #degrees
        step_per_sec: 50
        speed: 2.0
        acceleration: 0.46
        torque: 40
        max_torque: 40
        offset: 0.0
        offset_gain: 0.0

    prismatic_joint_names: ['finger_joint']
    prismatic_joints:
      finger_joint:
        index: 0
        id: 16
        invert: False
        min_pos: 0.0
        max_pos: 0.042
        horn_radius: 0.014
        angle_resolution: 0.0879 #degrees
        step_per_sec: 50
        speed: 0.5 
        acceleration: 2
        torque: 3
        max_torque: 40
        acceleration: 2
        offset: 0.0
        offset_gain: 0.0